docs/model/datamodel.md

```diff
@@ -0,0 +1,342 @@
+
+### ID
+
+Graphback requires each model to have an ID field, which is used to uniquely represent every object in the database.
+
+By default, Graphback expects an ID field of `id: ID`.
+
+```graphql
+"""@model"""
+type User {
+  id: ID!
+  username: String!
+}
+```
+
+You can set a custom ID field with the `@id` annotation.
+
+```graphql
+"""@model"""
+type User {
+  """@id"""
+  username: String!
+}
+```
+
+### @index
+
+The `@index` annotation can be used with [GraphQL Migrations](../databases/postgres) to create indexes on your PostgreSQL database tables.
+
+You can create one or multiple indexes per table.
+
+```graphql
+"""@model"""
+type User {
+  id: ID!
+  """
+  @index
+  """
+  name: String!
+  """
+  @index(name: 'userEmail_Idx', type: 'hash')
+  """
+  email: String!
+}
+```
+
+This creates two custom indexes on the `user` table:
+
+```
+"user_name_index" btree (name)
+"userEmail_index" hash (email)
+ ```
+
+You can also create a composite index.
+
+```graphql
+"""@model"""
+type User {
+  id: ID!
+  """
+  @index(name: 'user_unique_index')
+  """
+  email: String!
+  """
+  @index(name: 'user_unique_index')
+  """
+  name: String!
+}
+```
+
+This creates the following composite index:
+
+```
+"user_unique_index" btree (email, name)
+```
+
+#### Arguments
+
+| Argument | Description | Example |
+|-|-|-|
+|`name`| Specifies the name of index. **Optional**. Defaults to `<tablename>_<columnname>_index`. Accepts a string value | `@index(name: 'myIndex'`)|
+|`type`| Specifies the index type. **Optional**. Accepts a string value. Supported values: `btree` (default), `hash`, `gist`, `spgist`, `gin` and `brin` | `@index(type: 'hash')`|
+
+### @db
+
+GraphQL Migrations is a library to perform database migrations tables for PostgreSQL using a GraphQL schema. Most of the migration operations can be specified with the `@db` annotation. Check out the docs for [GraphQL Migrations](../docs/databases/migrations) to learn more.
+
+### @default
+
+The `@default` annotation is used to specify a default value. This is currently only supported for PostgreSQL through [GraphQL Migrations](../docs/databases/migrations) package.
+
+```graphql
+"""@model"""
+type User {
+  id: ID!
+  """@default(value: false)"""
+  verified: Boolean!
+}
+```
+
+#### Arguments
+
+| Argument | Description | Example |
+|-|-|-|
+|`value`| Specifies the default value for the field. **Required**. | `@default(value: 1)`|
+
+## MongoDB
+
+### @index
+
+The `@index` annotation can be used to create an index on a specific field or a set of fields at runtime.
+
+The arguments provided to `@index` are parsed into an [Index Definition](https://mongodb.github.io/node-mongodb-native/3.5/api/Collection.html#~IndexDefinition) object and then passed to [db.collection.createIndexes](https://mongodb.github.io/node-mongodb-native/3.5/api/Collection.html#createIndexes). 
+
+From following example data model:
+
+```graphql
+"""
+@model
+"""
+type Comment {
+  id: ID!
+  """
+  @index
```

Comment:
```suggestion
  Add an index with generated name 
  @index
```

I'll add a comment about what this does since it seems to be missing in this section